aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -131,6 +170,8 @@
   private String glueCatalogId;
   private boolean glueCatalogSkipArchive;
 
+  private AwsClientFactory clientFactory;
```

Comment:
I don't feel like the factory reference should be held by `AwsProperties`.  This also complicates the Serialization as AwsClientFactory now needs to be Serializable and I'm not even sure that it is at this point due to the `UrlConnectionHttpClient.create()` reference.  It seems like we should separate `AwsProperties` and the `AwsClientFactory`.

Reply:
`UrlConnectionHttpClient.create()` is a static final, should not affect serialization.

I think the client factory has to be serializable anyway, because `AwsClientFactory::s3`will be used as the supplier for s3 client in `S3FileIO`. Is there any use case that you think need it to be not serializable?